### PR TITLE
ci(*): add brigade

### PR DESCRIPTION
**Pull request by vdice**

### CODE REVIEW

File: brigade.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+// **********************************************
+// Event Handlers
+// **********************************************
+
+events.on("check_suite:requested", runSuite);
+events.on("check_suite:rerequested", runSuite);
+events.on("check_run:rerequested", runSuite);
+
+events.on("exec", (e, p) => {
```
Comment (carolynvs) [inline brigade.js]: Can you remind me what this does vs the runSuite function? I noticed that if I just do the exec event, it's not enough and I'm wondering what this exec event does.

File: brigade.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+class GoJob extends Job {
+  constructor (name) {
+    super(name);
+
+    const gopath = "/go";
+    const localPath = gopath + `/src/github.com/${projectOrg}/${projectName}`;
+
+    // Here using the large-but-useful deis/go-dev image as we have a need for deps
+    // already pre-installed in this image, e.g. helm, az, docker, etc.
+    // TODO: replace with lighter-weight image (Carolyn)
```
Comment (carolynvs) [inline brigade.js]: omg I really do owe you this... 🤦‍♀️

File: brigade.js (hunk 1 above)
Comment (vdice) [reply brigade.js]: Certainly!  The `exec` event is the default event that the `brig` cli sends on a `brig run <brigade project>` invocation.  Thus, I like having a handler for this event for easy testing locally on a given brigade project, e.g., `brig run vdice/porter-terraform`.

Meanwhile, the `runSuite` function is invoked when the `check_suite/run:*` events come in via GitHub on appropriate GH events (a push to master, a pull request, etc.).  As the functions that `runSuite` calls require appropriate webhook data (branch name, author data, etc. -- all of the usual data that GH POSTs), it isn't conducive to testing locally.

Does this help answer the question?

File: brigade.js (hunk 2 above)
Comment (vdice) [reply brigade.js]: :) We'll get to it in our copious free time :)

File: brigade.js (hunk 1 above)
Comment (carolynvs) [reply brigade.js]: Yup! Thanks that makes sense.

**Pull request merged**